Update workflow docs (#879)
kyma-project/btp-manager: docs/contributor/04-10-workflows.md

## E2E Tests Workflow 

> This workflow uses the DEV artifact registry, tags the binary image and OCI module image with the PR number, and calls the reusable [workflow](/.github/workflows/run-e2e-tests-reusable.yaml).

Reviewer: ```suggestion
This workflow uses the DEV artifact registry, tags the binary image and OCI module image with the PR number, and calls the [reusable workflow](/.github/workflows/run-e2e-tests-reusable.yaml).
```

## Unit Tests Workflow

> This workflow calls the reusable [workflow](/.github/workflows/run-unit-tests-reusable.yaml).

Reviewer: ```suggestion
This workflow calls the [reusable workflow](/.github/workflows/run-unit-tests-reusable.yaml).
```

## Markdown Links Check Workflow

This [workflow](/.github/workflows/markdown-link-check.yaml) is triggered daily at midnight and by each PR on the `main` branch. It checks for dead links in the repository.

## Govulncheck Workflow

This [workflow](/.github/workflows/run-govulncheck.yaml) runs the Govulncheck.

## Auto Merge Workflow

This [workflow](/.github/workflows/auto-merge.yaml) enables the auto-merge functionality on a PR that is not a draft.

## All Cheks Passed Workflow

> This [workflow](/.github/workflows/pr-checks.yaml) checks if all jobs, expect those that are excluded in the workflow configuration, have passed.

Reviewer: ```suggestion
This [workflow](/.github/workflows/pr-checks.yaml) checks if all jobs, except those excluded in the workflow configuration, have passed.
```
